File: src/renderer/src/types/sftp.ts
export interface FileInfo {
  name: string
  path: string
  size: number
  mode: number
  mod_time: number
  is_dir: boolean
}

export interface TransferProgress {
  transfer_id: string
  filename: string
  total: number
  transferred: number
  percentage: number
  status: 'uploading' | 'downloading' | 'completed' | 'failed'
}

export interface ListRequest {
  path: string
}

export interface UploadRequest {
  local_path: string
  remote_path: string
}

export interface DownloadRequest {
  remote_path: string
  local_path: string
}

export interface DeleteRequest {
  path: string
}

export interface MkdirRequest {
  path: string
}

export interface RenameRequest {
  old_path: string
  new_path: string
}

export interface RemoteTransferResult {
  source_path: string
  dest_path: string
  success: boolean
  error?: string
}

export interface RemoteTransferProgress {
  total_items: number
  completed_items: number
  failed_items: number
  current_item: string
  bytes_transferred: number
  total_bytes: number
}

File: backend/internal/models/sftp.go
package models

type FileInfo struct {
	Name    string `json:"name"`
	Path    string `json:"path"`
	Size    int64  `json:"size"`
	Mode    uint32 `json:"mode"`
	ModTime int64  `json:"mod_time"`
	IsDir   bool   `json:"is_dir"`
}

type TransferProgress struct {
	TransferID string  `json:"transfer_id"`
	Filename   string  `json:"filename"`
	Total      int64   `json:"total"`
	Transferred int64  `json:"transferred"`
	Percentage float64 `json:"percentage"`
	Status     string  `json:"status"` // uploading, downloading, completed, failed
}

type ListRequest struct {
	Path string `json:"path"`
}

type UploadRequest struct {
	LocalPath  string `json:"local_path"`
	RemotePath string `json:"remote_path"`
}

type DownloadRequest struct {
	RemotePath string `json:"remote_path"`
	LocalPath  string `json:"local_path"`
}

type DeleteRequest struct {
	Path string `json:"path"`
}

type MkdirRequest struct {
	Path string `json:"path"`
}

type RenameRequest struct {
	OldPath string `json:"old_path"`
	NewPath string `json:"new_path"`
}

type CancelRequest struct {
	TransferID string `json:"transfer_id"`
}

type ReadFileRequest struct {
	Path string `json:"path"`
}

type ReadFileResponse struct {
	Content string `json:"content"`
	Path    string `json:"path"`
}

type WriteFileRequest struct {
	Path    string `json:"path"`
	Content string `json:"content"`
}
